.github/tsc.json

```diff
@@ -4,14 +4,15 @@
             "owner": "tsc",
             "pattern": [
                 {
-                    "regexp": "^(?:\\s+\\d+\\>)?([^\\s].*)\\((\\d+|\\d+,\\d+|\\d+,\\d+,\\d+,\\d+)\\)\\s*:\\s+(error|warning|info)\\s+(\\w{1,2}\\d+)\\s*:\\s*(.*)$",
+                    "regexp": "^(?:\\s+\\d+\\>)?([^\\s].*)\\((\\d+)(?:,(\\d+)(?:,\\d+(?:,\\d+)?)?)?\\)\\s*:\\s+(error|warning|info)\\s+(\\w{1,2}\\d+)\\s*:\\s*(.*)$",
                     "file": 1,
-                    "location": 2,
-                    "severity": 3,
-                    "code": 4,
-                    "message": 5
+                    "line": 2,
+                    "column": 3,
+                    "severity": 4,
+                    "code": 5,
+                    "message": 6
```

Comment:
I cannot human-parse this regex, but maybe this should just use the official `$tsc` pattern from vscode? Source: https://github.com/microsoft/vscode/blob/master/extensions/typescript-language-features/package.json#L974-L981 (it is not the same as this one)

Reply:
I have tested the official regex and can confirm that it works:

```js
// commit this file to .github/tsc.json
{
  "problemMatcher": [
    {
      "owner": "tsc",
      "pattern": [
        {
          "name": "tsc",
          "regexp": "^([^\\s].*)[\\(:](\\d+)[,:](\\d+)(?:\\):\\s+|\\s+-\\s+)(error|warning|info)\\s+TS(\\d+)\\s*:\\s*(.*)$",
          "file": 1,
          "line": 2,
          "column": 3,
          "severity": 4,
          "code": 5,
          "message": 6
        }
      ]
    }
  ]
}
```

```yaml
- name: Set up Node
  uses: actions/setup-node@v1
# Run this _after_ setup-node
- name: Set up tsc problem matcher
  run: echo "::add-matcher::${GITHUB_WORKSPACE}/.github/tsc.json"
```

Reply:
Here is my regex, free-spaced and syntax highlighted:

![image](https://user-images.githubusercontent.com/551184/83219319-e9b3f100-a124-11ea-8fd9-dad4527cca96.png)

Here is the official pattern from TypeScript, free-spaced and syntax-highlighted:

![image](https://user-images.githubusercontent.com/551184/83219042-28957700-a124-11ea-98d7-19f7ae5e8d90.png)

A TypeScript error message looks like this:

```
lib/net.ts(48,26): error TS1005: '}' expected.
```

Both regexes support that pattern. Mine is modified from the old one in this repository, which supported a variable number of commas.

In "fancy" mode, a TypeScript error message looks like this:

```
lib/net.ts:48:26 - error TS1005: '}' expected.

48  			out += `${key}=${enc odeURIComponent('' + body[key])}`;
    			                     ~~~~~~~~~~~~~~~
```

The official regex also supports this format (although it should never be passed to GitHub Actions in this format, in practice).

Reply:
In conclusion, the official pattern is probably better.

Do you think the free-spaced annotated version should be put somewhere, for readability?

```
^
# filename
([^\s].*)
# line number in "(1,2)" or ":1:2 -" format
[\(:]
(\d+) [,:] (\d+)
(?:\):\s+|\s+-\s+)
# severity
(error|warning|info)
# error code
\s+ TS(\d+)
# ":"
\s* :
# error message
\s* (.*)
$
```

(It can be compiled into the version in the repository with https://zarel.github.io/regexfree/ )

Reply:
(One other difference is that the official pattern doesn't consider "TS" part of the error code. There's an argument that it should be kept, for googleability.)